docs_src/getting-started/Ch-GettingStartedSnapUsers.md

```diff
@@ -6,170 +6,730 @@
+    ```
+
+#### Setting TLS certificates
+The API Gateway setup generates a self-signed certificate by default. To replace that with your own certificate, refer to API Gateway guide: [Using a bring-your-own external TLS certificate for API gateway](../../security/Ch-APIGateway/#using-a-bring-your-own-external-tls-certificate-for-api-gateway) and use the snapped `edgexfoundry.secrets-config` utility.
+
+To get the usage help:
 ```bash
-sudo snap set edgexfoundry support-notifications=off
+edgexfoundry.secrets-config proxy tls -h
 ```
+You may also refer to the [secrets-config proxy](../../security/secrets-config-proxy/) documentation.
+
+!!! example 
+    Given certificate `cert.pem`, private key `privkey.pem`, and certificate authority `ca.pem` files:
+
+    Read the API Gateway token:
+    ```bash
+    KONG_ADMIN_JWT=`sudo cat /var/snap/edgexfoundry/current/secrets/security-proxy-setup/kong-admin-jwt`
+    ```
+
+    Add the certificate, using Kong Admin JWT to authenticate:
+    ```bash
+    edgexfoundry.secrets-config proxy tls --incert cert.pem --inkey privkey.pem --admin_api_jwt $KONG_ADMIN_JWT
+    ```
+
+    Try it out:
+    ```bash
+    curl --cacert ca.pem https://localhost:8443/core-data/api/v2/ping
+    ```
+    Output: `{"message":"Unauthorized"}`  
+    This means that TLS is setup correctly, but the request is not authorized.  
+    Set the `-v` command for diagnosing TLS issues.
 
-All services which are installed on the system as systemd units, which if enabled will automatically start running when the system boots or reboots.
 
-### Configuring individual services
-This snap supports configuration overrides via snap configure hooks which generate service-specific .env files which are used to
-provide a custom environment to the service, overriding the default configuration provided by the service's `configuration.toml`
-file. If a configuration override is made after a service has already started, then the service must be **restarted** via command-line
-(e.g. `snap restart edgexfoundry.<service>`), or [snapd's REST API](https://snapcraft.io/docs/snapd-api). If the overrides are provided via the snap configuration defaults
-capability of a gadget snap, the overrides will be picked up when the services are first started.
+??? tip "Seeding a custom TLS certificate using snap options"
+    To spin up an EdgeX instance with a custom certificate, the snap provides the following configuration options:
+    
+    * `apps.secrets-config.proxy.tls.cert`
+    * `apps.secrets-config.proxy.tls.key`
+    * `apps.secrets-config.proxy.tls.snis` (comma-separated values)
 
-The following syntax is used to specify service-specific configuration overrides for the edgexfoundry snap:
+    This is particularly useful when seeding the snap from a [Gadget](https://snapcraft.io/docs/gadget-snap) on an [Ubuntu Core](https://ubuntu.com/core) system.
 
+#### Secret Store token interface
```

Comment:
services' readmes will be redirected to `secret store token`:
[secret-store-token]: https://docs.edgexfoundry.org/2.2/getting-started/Ch-GettingStartedSnapUsers/#secret-store-token

Reply:
Hmm.. I added the interface without thinking about the side effects. Will revert it.